mpack-instance-manager/src/main/python/instance_manager/mpack-instance-manager.py

```diff
@@ -152,25 +157,24 @@ def main(options, args):
                        components_map=parsed_components_map)
 
   elif action == GET_CONF_DIR_ACTION:
-    print get_conf_dir(mpack=options.mpack, mpack_instance=options.mpack_instance,
+    print json.dumps(get_conf_dir(mpack=options.mpack, mpack_instance=options.mpack_instance,
                        subgroup_name=options.subgroup_name, module_name=options.module_name,
-                       components_map=parsed_components_map)
+                       components_map=parsed_components_map), indent=indent)
 
   elif action == GET_LOG_DIR_ACTION:
-    print get_log_dir(mpack=options.mpack, mpack_instance=options.mpack_instance,
+    print json.dumps(get_log_dir(mpack=options.mpack, mpack_instance=options.mpack_instance,
                        subgroup_name=options.subgroup_name, module_name=options.module_name,
-                       components_map=parsed_components_map)
+                       components_map=parsed_components_map), indent=indent)
 
   elif action == GET_RUN_DIR_ACTION:
-    print get_run_dir(mpack=options.mpack, mpack_instance=options.mpack_instance,
+    print json.dumps(get_run_dir(mpack=options.mpack, mpack_instance=options.mpack_instance,
                        subgroup_name=options.subgroup_name, module_name=options.module_name,
-                       components_map=parsed_components_map)
+                       components_map=parsed_components_map), indent=indent)
 
   elif action == LIST_INSTANCES_ACTION:
-    print list_instances(mpack=options.mpack, mpack_instance=options.mpack_instance,
-                         subgroup_name=options.subgroup_name, module_name=options.module_name,
-                         components_map=parsed_components_map)
-
+    print json.dumps(list_instances(mpack=options.mpack, mpack_instance=options.mpack_instance,
```

Comment:
We should simply update instance_manager.py::build_json_output() function to indent the json output. We shouldn't make changes to mpack_instance_manager.py to do this indentation after every function call. Please back out these indentation changes.

Reply:
@jayush From programming point of view, indent is useless, it is only used to display readable output for the user, if we choose tree like output, the indent will be removed. instance_manager.py::build_json_output() only returns a python dict, I do not think indent  can be used here, I assume  indent can only apply to json.dumps.